### PR TITLE
Bugfix/JSSNMSHDD-2771 do not throw in usecase

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

File: src/useCases/common/UseCase.ts [hunk 1]
```diff
@@ -21,12 +21,16 @@ export abstract class UseCase<IRequest, IResponse> {
...
+            return this.resultFromUnknown(e);
+        }
+    }
 
+    private resultFromUnknown(e: unknown) {
+        if (e instanceof Error) {
             return this.resultFromError(e);
         }
+
+        return Result.fail(new ApplicationError("error.runtime.unknown", `An unknown object was thrown in a UseCase: ${JSON.stringify(e)}`));
```
Comment (stnmtz) [inline src/useCases/common/UseCase.ts:33]: Hmm, JSON.stringify could cause other errors if recursion is in the object.

File: src/useCases/common/UseCase.ts [hunk 2]
```diff
@@ -21,12 +21,16 @@ export abstract class UseCase<IRequest, IResponse> {
...
             return await this.executeInternal(request);
         } catch (e) {
-            if (!(e instanceof Error)) {
-                throw e;
-            }
+            return this.resultFromUnknown(e);
+        }
+    }
 
+    private resultFromUnknown(e: unknown) {
```
Comment (stnmtz) [inline src/useCases/common/UseCase.ts:28]: failFromUnknown?

File: src/useCases/common/UseCase.ts [hunk 3]
```diff
@@ -38,7 +42,11 @@ export abstract class UseCase<IRequest, IResponse> {
             return this.handleServalError(error);
         }
 
-        throw error;
+        if (error instanceof ApplicationError) {
+            return Result.fail(error);
+        }
+
+        return Result.fail(new ApplicationError("error.runtime.unknown", `An error was thrown in a UseCase: ${error.message}`, error));
```
Comment (stnmtz) [inline src/useCases/common/UseCase.ts:49]: RuntimeErrors.general.unknown(message)

File: src/useCases/common/UseCase.ts (hunk 1 above)
Comment (stnmtz) [reply src/useCases/common/UseCase.ts:33]: RuntimeErrors.general.unknown(error)

**Review by stnmtz — CHANGES_REQUESTED** [review]

File: src/useCases/common/UseCase.ts (hunk 2 above)
Comment (jkoenig134) [reply src/useCases/common/UseCase.ts:28]: it returns a result

File: src/useCases/common/UseCase.ts (hunk 3 above)
Comment (jkoenig134) [reply src/useCases/common/UseCase.ts:49]: Doesn't exist

**Review by stnmtz — APPROVED** [review]

**Pull request merged**